src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbolBase.cs

```diff
@@ -142,28 +165,49 @@ private enum Flags : byte
                 _name = _lazySourceName = memberName;
             }
 
-            if ((isAutoProperty && hasGetAccessor) || hasInitializer)
+            if (isAutoProperty || hasInitializer)
```

Comment:
>hasInitializer [](http://example.com/codeflow?start=34&length=14)

The current specification says the following: "An auto-property may optionally have a property_initializer, which is applied directly to the backing field as a variable_initializer." It looks like the current proposal doesn't change that. However, it feels like it would be unexpected to apply the initializer directly to the backing field when there is a set accessor with a body. It feels like the accessor should be invoked instead then.

Reply:
@AlekseyTs I'm reading the comment as if it's related to what code will be synthesized for the initializer, not sure why this comment is specifically about this line?

Reply:
I'm also not sure if you mean the spec doesn't make sense to you?

Also consider this `public int P { get => field; } = 5;`, what should be its behavior?

Reply:
> I'm reading the comment as if it's related to what code will be synthesized for the initializer, 

That is correct.

> not sure why this comment is specifically about this line?

Sorry for the confusion. I should have made it clear that it is not specifically about this line. I couldn't find a "better" modified line that has some relation to the issue.

> I'm also not sure if you mean the spec doesn't make sense to you?

I am saying that the behavior as specified feels wrong to me.

>Also consider this ```public int P { get => field; } = 5;```, what should be its behavior?

Since this property doesn't have declared setter, the specified behavior (initialize the backing field) is reasonable. Because there isn't really other alternative with exception of disallowing this. The behavior would be reasonable for ```public int P { get => field; set; } = 5;``` because this is what the generated setter would do anyway.

However, the specified behavior (initialize the backing field) doesn't feel reasonable for this scenario:
```
public int AlwaysEven { get => field; set {field = value * 2} } = 5;
```

Reply:
https://github.com/dotnet/csharplang/pull/5573 brings this to LDM.